documentation/usage/errorcodes/index.md

```diff
@@ -16,6 +16,38 @@ These error codes may appear from any command, and are indicative of more genera
 - **Caused by:** An unexpected error within Flyway (e.g. a null pointer exception)
 - **Solution:** Please contact support or create a [GitHub issue](https://github.com/flyway/flyway/issues)
 
+### `ERROR`
+- **Caused by:** An unexpected error within Flyway (e.g. a null pointer exception)
```

Comment:
When we first designed this feature, we had in mind the distinction between an error and a fault.

A fault is an unexpected failure - like a bug. The software has gone wrong even with valid input. On the other hand, an error is a failure state due to invalid user input (e.g. providing invalid schema name).

We used this as inspiration: https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#5-taxonomy

(Most of this context might be lost / buried in github issues, but this is what I remember).

Now perhaps the code itself doesn't abide by this distinction, but it ought to.

Reply:
Ah yes, here is the issue: https://github.com/red-gate/flyway-main/pull/53#issue-513350304

> My idea is to release something simple and gather feedback from the Orcas. We can then adjust this feature based on the feedback.

Well it took over a year, and a different team. But we got feedback eventually 😅

Reply:
see latest commit